subsys/net/openthread/settings_check/CMakeLists.txt

```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+# This script checks if NVS settings backend is used with nRF54L SoC target and prints
+# the warning that the recommended backend is ZMS due to the RRAM design.
+
+if (CONFIG_SOC_SERIES_NRF54LX AND CONFIG_SETTINGS_NVS)
+    message(WARNING " ###################################################################################\n"
+                    " #                                                                                 #\n"
+                    " # Your application uses NVS backend for the settings storage that is not          #\n"
```

Comment:
get rid of the excess spacing on the right here. Also seems a bit odd that this is being added for openthread?

Reply:
Why is it odd? This is because the Zephyr OpenThread L2 layer implies NVS and we cannot fix it reasonably fast now. The ZMS was enabled in all OpenThread samples, but still someone may have an own application copy that will not be automatically aligned, so the check will be done for all applications running OpenThread.

Reply:
It should be a global thing for any code so users can know if they've got something set up that will kill their device in half the time as one would expect

Reply:
Yeah, I agree, but I was asked only to handle integration with OpenThread samples, so I've taken care only about this use case. Maybe it could be moved and used in some generic place if others agree on that.